From 540ce286d3aa29c1656d15c9c75521709062d6ac Mon Sep 17 00:00:00 2001
From: Jason Montleon <jmontleo@redhat.com>
Date: Thu, 18 Dec 2025 14:50:51 -0500
Subject: [PATCH 0001/1223] Revert "clk: spacemit: Set clk_hw_onecell_data::num
 before using flex array"

This reverts commit 251e09ca0ce4df6860347eaa193b6115ce46bcd1.
---
 drivers/clk/spacemit/ccu-k1.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/clk/spacemit/ccu-k1.c b/drivers/clk/spacemit/ccu-k1.c
index 7c9585c9359de..65e6de030717a 100644
--- a/drivers/clk/spacemit/ccu-k1.c
+++ b/drivers/clk/spacemit/ccu-k1.c
@@ -973,8 +973,6 @@ static int spacemit_ccu_register(struct device *dev,
 	if (!clk_data)
 		return -ENOMEM;
 
-	clk_data->num = data->num;
-
 	for (i = 0; i < data->num; i++) {
 		struct clk_hw *hw = data->hws[i];
 		struct ccu_common *common;
@@ -1001,6 +999,8 @@ static int spacemit_ccu_register(struct device *dev,
 		clk_data->hws[i] = hw;
 	}
 
+	clk_data->num = data->num;
+
 	ret = devm_of_clk_add_hw_provider(dev, of_clk_hw_onecell_get, clk_data);
 	if (ret)
 		dev_err(dev, "failed to add clock hardware provider (%d)\n", ret);
-- 
2.52.0

